### PR TITLE
ci: Switch from `hatch`, `pip` to `uv` in GitHub Actions

### LINKED ISSUE #3718: Switch from `hatch`, `pip` to `uv` in GitHub Actions
### What is your suggestion?

 _Originally posted by @dangotbanned in [#3577](https://github.com/vega/altair/issues/3577#issuecomment-2334214226)_

A first step would be trying to replace `pip` and `hatch` in all the GitHub workflows.
They now have proper documentation - which was missing [when I tried](https://github.com/vega/altair/pull/3431#issuecomment-2168654915) that out before

### Have you considered any alternative solutions?

_No response_

### PR DESCRIPTION
**Pull request by dangotbanned**
Closes #3718

One step towards #3577

This PR moves to `uv` in each of:
- [`build.yml`](https://github.com/dangotbanned/altair/blob/8d8c95254ca73c37e8e791307cca625071c107ed/.github/workflows/build.yml)
- [`docbuild.yml`](https://github.com/dangotbanned/altair/blob/8d8c95254ca73c37e8e791307cca625071c107ed/.github/workflows/docbuild.yml)
- [`lint.yml`](https://github.com/dangotbanned/altair/blob/8d8c95254ca73c37e8e791307cca625071c107ed/.github/workflows/lint.yml)

I tried this a while back, but gave up on (https://github.com/vega/altair/pull/3719/commits/27999c1b041cd2f685dc05d37cf613eb2dfc4020).
Since then, I've used `uv` some more as I mentioned in (https://github.com/vega/altair/issues/3577#issuecomment-2548280136) - which helped unblock this part 😅

### CODE REVIEW

File: .github/workflows/docbuild.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ name: docbuild
 
 on: [push, pull_request]
 
+env:
+  UV_SYSTEM_PYTHON: 1
```
Comment (mattijn) [inline .github/workflows/docbuild.yml:6]: Great PR @dangotbanned! Do you have a reference why this is environment definition is necessary? Thanks!

**Comment by mattijn** [comment]
Regarding feedback on parallel running of sphinx docs:

```
WARNING: the sphinxext_altair.altairplot extension does not declare if it is safe for parallel reading, assuming it isn't - please ask the extension author to check and make it explicit
```

Is this something worthwhile pursuing for?

File: .github/workflows/docbuild.yml (hunk 1 above)
Comment (dangotbanned) [reply .github/workflows/docbuild.yml:6]: Thanks @mattijn!
Yeah I can explain.

I've enabled [`UV_SYSTEM_PYTHON`](https://docs.astral.sh/uv/configuration/environment/#uv_system_python) as a [workflow environment variable](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#defining-environment-variables-for-a-single-workflow) to add the `--system` argument on all relevant `uv` commands.
You can see in [`narwhals/.github/workflows/downstream_tests.yml`](https://github.com/narwhals-dev/narwhals/blob/7316b2eaa35801559876f42bc911d1285f28c3ef/.github/workflows/downstream_tests.yml) that argument is [used **25** times](https://github.com/search?q=repo%3Anarwhals-dev%2Fnarwhals%20path%3A.github%2Fworkflows%2Fdownstream_tests.yml%20%22--system%22&type=code). 

So this is just to keep it simple, making sure it is there when needed - but isn't something you need to worry about if updating the workflow in the future 🙂

**Comment by dangotbanned** [comment]
> Regarding feedback on parallel running of sphinx docs:
> 
> ```
> WARNING: the sphinxext_altair.altairplot 
> extension does not declare if it 
> is safe for parallel reading, 
> assuming it isn't - please ask 
> the extension author to check 
> and make it explicit
> ```
> 
> Is this something worthwhile pursuing for?

@mattijn 

It has the *potential* to speed this CI job up, but I think I'd struggle with testing [`--jobs`](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-j) locally:

> This feature only works on systems supporting “fork”. Windows is not supported. 

I'm happy to support if anyone else wants to take the lead on it though. 
I just imagine getting it working on native `posix` will be easier than [`wsl`](https://learn.microsoft.com/en-us/windows/wsl/install) or [`MINGW64`](https://www.mingw-w64.org/)

File: .github/workflows/docbuild.yml (hunk 1 above)
Comment (dangotbanned) [reply .github/workflows/docbuild.yml:6]: I really need to transition away from windows at some point 😔

Think it is causing problems in https://github.com/vega/vega-datasets/pull/648#issuecomment-2558086569 as well

**Comment by mattijn** [comment]
Thanks for providing context regarding the environment variable of `uv`!

In fact I'm quite happy that we have maintainers on multiple operating systems as it increase the robustness of the codebase.

**Pull request merged**